datasetUtils/getUserInfoFromToken.go

```diff
@@ -13,42 +13,53 @@ type UserInfo struct {
 	CurrentGroups    []string `json:"currentGroups"`
 }
 
-func GetUserInfoFromToken(client *http.Client, APIServer string, token string) (map[string]string, []string) {
+/* GetUserInfoFromToken makes a GET request to the provided APIServer with the provided token to fetch user information.
+
+Parameters: client: An *http.Client object to make the HTTP request. APIServer: A string representing the URL of the API server. token: A string representing the user's access token.
+
+Returns: A map[string]string where the keys are "username", "mail", "displayName", and "accessToken", and the values are the corresponding user information. A slice of strings representing the groups the user is a member of.
+
+If the HTTP request fails or the response status code is not 200, the function will log the error and terminate the program. If the user information cannot be unmarshalled into the UserInfo struct or the user cannot be mapped to the token, the function will log the error and terminate the program. */
+func GetUserInfoFromToken(client *http.Client, APIServer string, token string) (map[string]string, []string, error) {
 	u := make(map[string]string)
 	accessGroups := make([]string, 0)
-
+	
+	// url := APIServer + "/Users/userInfos" # uncomment if you use the Authorization header method
 	url := APIServer + "/Users/userInfos?access_token=" + token
 	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create request: %w", err)
+	}
+	
 	req.Header.Set("Content-Type", "application/json")
-
+	// req.Header.Set("Authorization", "Bearer "+token)  # NOTE: this is a more secure method, but I am not sure whether the server accepts tokens in the Authorization header.
```

Comment:
I would leave it as is for now (i.e. in the query params), and remove the comment. But we can indeed open an issue for this and do in a later PR. I don't think at the moment that's supported (at least not the Bearer prefix)

Reply:
ok